### DISCUSSION TITLE
C++11 compatibility

### DISCUSSION THREAD

**Discussion by dimitre** [openframeworks/openFrameworks#7414] [General]
Why it is important? 
I understand some macOS can't use c++17 so OF is still backwards compatible
Any other reason to keep everything compatible with c++11 ?

**Comment by ofTheo** [comment]
I think some people making software to be distributed might want to be able to support older machines, or people using an older linux distro, but I think at this point we should aim for C++17 and only support C++11 if its not too costly to the project.

**Reply by ofTheo** [reply]
@dimitre This thread with @himwho provides some context: 
https://github.com/openframeworks/openFrameworks/issues/6585#issuecomment-1145302683 

And then there is this post from @roymacdonald who is still rocking 10.14.6  🙂
https://github.com/openframeworks/openFrameworks/issues/6585#issuecomment-1101959729

**Reply by roymacdonald** [reply]
Hi @ofTheo I've been using macos 11.6 for some time now. 

My 2 cents. I think that deprecation has to happen at some point and if keeping compatibility with older version has a cost to the project then I think it is reasonable to stop supporting those. Besides, if someone want to use an older version of C++ they can also use an older version of OF. Maybe it can be helpful to state somewhere, maybe the downloads section which is the last version that supports c++11

**Reply by dimitre** [reply]
Thanks. maybe as a small step we can bump to c++14 and remove boost filesystem and make_unique template soon, maybe for 0.12.0 what do you think? I think 14 should support some fairly old machines too

**Comment by 2bbb** [comment]
I think, C++17 is quite become common.
I look some or many modern libraries (but not so major) are only support after C++17.

and I think, 10.14.6 users who are particular about the that OS can use 11.2.
but maybe users who uses most modern macOS (e.g. 10.15, 11, 12) are happy if who can use most modern API.
if we will follow old OS users, I think better to release patch release on 11.3 or later like LTS. (I know if we do those then we need to pay some costs.)

nevertheless, good point of openFrameworks (what I think) is easy to use many C++ libraries.
but currently, only some users can use those who can fix codes for C++17 or later.

sorry,I don't know good closing words for this topic in english. but I think so.

**Comment by artificiel** [comment]
FYI QtCreator bumped to C++17 on all platforms #7461
